src/sequencer.c

```diff
@@ -0,0 +1,227 @@
+// timer.c 
+// handle timer-like events for music
+// runs in a thread / task
+
+/* 
+AMY style sequencer spec
+for now: fixed ppq (48)
+
+you can pass "sequence" / H as a string,up to 3 values, tick,divider,tag 
+if divider is 4, then it's every 4 ticks (48 ppq at bpm)
+tag is a numerical constant, can be whatever (uint32)
+sequence=",4,123" -- no tick, just divider, tag = 123
+sequence is passed along other messages, but is treated like time=X  -- not played until the sequence fires
+if sequence="59320,,123" then this is an absolute tick in the future and played only once
+you can remove scheduled sequences with sequencer=",,123" -- nothing in the tick section will delete a message if there is a tag
+tick takes precedence over divider if both given
+
+ok, how we deal with it internally:
+
+in the parse_amy_message, we check for sequence, if it's there, we set the event type to EVENT_SEQUENCE and send it to
+sequence_add_event(struct event e, tick, divider, tag) 
+
+sequence_add_event checks / parses the sequence string and fills in a new entry in the seqeunce entry LL
+it also can remove sequence entries if no tick is given and the tag exists 
+
+a thread/timer that checks every 48ppq tick with BPM 
```

Comment:
This appears to be the start of a new, perhaps older description. Punctuation can help interpret.